src/view/actions/directCall.jsx

```diff
@@ -11,41 +11,194 @@
+                />
+                <ActionButton
+                  aria-label="Delete"
+                  isQuiet
+                  onPress={() => {
+                    fields.remove(index);
+                    if (fields.length === 1) {
+                      fields.push({});
+                    }
+                  }}
+                >
+                  <CloseIcon size="XS" />
+                </ActionButton>
+              </Flex>
+            ))}
+            <span className="codeLine">
+              {String.fromCharCode(CLOSING_CURLY)}
+              {String.fromCharCode(SEMICOLON)}
+            </span>
+            <span className="codeLine">
+              return detail{String.fromCharCode(SEMICOLON)}
+            </span>
+          </Well>
+        </View>
+        <View {...DEFAULT_BLANK_SPACE_PROPS} />
+      </Flex>
+      <ActionButton
+        marginTop="size-200"
+        onPress={() => fields.push({})}
+        aria-label="Add detail field"
+      >
+        Add Detail Field
+      </ActionButton>
+    </>
+  );
+};
+
+const filterEmptyDetailObjectRows = (detailObjectsArray) => {
+  if (!detailObjectsArray?.length) {
+    return detailObjectsArray;
+  }
+
+  return detailObjectsArray.reduce((entries, nextRow) => {
+    const { key, value } = nextRow;
+    if (nextRow.key?.length && nextRow.value?.length) {
+      return entries.concat({ key, value });
+    }
+    return entries;
+  }, []);
+};
 
 const DirectCall = () => (
-  <FullWidthField
-    label="Direct Call Identifier"
-    name="identifier"
-    containerMinWidth="size-6000"
-    isRequired
-  />
+  <div className="directCall">
+    <FullWidthField
+      label="Direct Call Identifier"
+      name="identifier"
+      containerMinWidth={containerMinWidth}
+      isRequired
+    />
+
+    <FieldArray component={DetailObjectRows} name="eventObjectEntries" />
+  </div>
 );
 
 export default DirectCall;
 
 export const formConfig = {
   settingsToFormValues(values, settings) {
+    let eventObjectEntries;
+    if (Array.isArray(settings.detail?.eventObjectEntries)) {
+      eventObjectEntries = settings.detail.eventObjectEntries;
+    } else {
+      eventObjectEntries = [{}];
+    }
+
     return {
-      ...values,
-      ...settings
+      ...settings,
+      eventObjectEntries
     };
   },
   formValuesToSettings(settings, values) {
-    return {
+    const eventObjectEntries = filterEmptyDetailObjectRows(
+      values.eventObjectEntries
+    );
+
+    const newSettings = {
       ...settings,
-      ...values
+      identifier: values.identifier
     };
+    if (eventObjectEntries.length) {
+      newSettings.detail = {
+        eventObjectEntries
+      };
+    }
+
+    return newSettings;
   },
   validate(errors, values) {
-    errors = {
-      ...errors
-    };
+    errors = { ...errors };
 
     if (!values.identifier) {
       errors.identifier = 'Please specify an identifier.';
     }
 
+    errors.eventObjectEntries = [];
```

Comment:
This validation is more strict than what you have defined in the JSON Schema. In the view the `value` key is required while in the JSON schema it is not. I think they should match. At the current state, using the API I can add settings objects that the view would consider them invalid.

Reply:
You could consider adding a validation that would check that the keys are unique. Having 2 keys with the same name would not break the functionality, but it would be nice to tell the user that he should clean up.